### PR TITLE
Js2

### LINKED ISSUE #60: Post Feed Observer To Load More Results

### PR DESCRIPTION
**Pull request by Anclagen**
This is my current working version I feel I have delivered most of the functionality to some degree.

Questions:
- At the moment I am grabbing all the posts for the search, and feed, then sorting/filtering the data, should I change this? and use sort/sort-by strings on the fetch request instead and make more requests as the user scrolls or changes the sort parameters.
-  I have the search results display in a separate section to the post feed, I assume this still meets this criteria? "User can search the post content feed".
- Should I try to break down some of my function further?  Especially in reference to my [post_cards.mjs](https://github.com/Anclagen/SoSocial/blob/JS2/js/modules/render/post_cards.mjsl) where I have a very large function to build my posts and add all the relevant listeners and functionality, I have split some of the code for the forms, and replies into other files.
- Error reporting with try catch blocks should I be placing these in the primary fetch functions or further down the chain where I use function to apply this to the page, or is it ok to nest multiple try catches by adding one at each stage?

### CODE REVIEW

**Comment by martink-rsa** [comment]
Hi Anclagen, could you add me as a reviewer please?

File: js/modules/render/loader.mjs [hunk 1]
```diff
@@ -0,0 +1,12 @@
+
+/**
+ * Adds loader to container
+ * @param {Element} container 
+ */
+export function addLoader(container){
+  container.innerHTML = `<div class="loader">
+                          <div class="outer-loader"></div>
+                          <div class="inner-loader"></div>
+                          <p>Getting products, please wait...</p>
```
Comment (martink-rsa) [inline js/modules/render/loader.mjs]: This loader states we are getting products; however, there aren't products on the page.

**Review by martink-rsa — CHANGES_REQUESTED** [review]
Hi @Anclagen,

Great work on the work so far! 

You've done all of the assignment requirements. 

I've left one comment in the code for you to change. 

There is also feedback below for suggested changes to make. Great job once again.

## Formatting issues:
Your code has formatting issues. Please install Prettier, set it as your default formatter, and then set it to `Format on Save` in the settings. This will ensure that your code is always formatted and consistent. If you need assistance with this, please get in touch with one of the tutors.

## Hard-coded URLs:
You have hard-coded URLs, such as in the `register` function. This makes it challenging to maintain your code. If the server domain were to change, you would need to change this throughout your code, making room for errors. You should instead have this set as a variable in a file that gets imported; then you add the path onto it. E.g.

```js
export const SITE_URL = “https://nf-api.onrender.com/api/v1/social”;
```

Then you’d use this in your API calls and add the path you need e.g.
```js
Import { SITE_URL } from ‘./constans.mjs’;

const url = `${SITE_URL}/auth/register`;
```

## Don’t use shortened names for variables:
You have a variable `qstring` which is likely to mean query string. Variable naming like this can be problematic and cost more time than you save by typing less. A developer, including yourself at a later stage in time, will need to try to figure out what the `q` means in `qstring`. Rather write out the whole variable name so that your variable’s name is clear.

## Remove unused code before final submission:
There is unused code in your submission, such as code that is commented our and/or imports that aren’t being used. I understand that the code is not complete yet, however make sure it gets cleaned up before your final submission.

## Consider redirecting the user to the site once they have registered an account:
Consider redirecting the user once they have successfully registered an account. You aren’t required to do this, but if you’re looking for a challenge, then it’s something you could do to improve the UX. 

## `post_cards.mjs` functions abstracted:

Your `post_cards.mjs` file contains a lot of code and different functions, leading to 400 lines of code.

You should consider abstracting parts of this file into new files. This will make your code cleaner and more testable.

Please don't hesitate to contact the tutors if you need any assistance with the coursework. 

Kind regards,
Martin

**Comment by martink-rsa** [comment]
Questions:

> 1. At the moment I am grabbing all the posts for the search, and feed, then sorting/filtering the data, should I change this? and use sort/sort-by strings on the fetch request instead and make more requests as the user scrolls or changes the sort parameters.

The way you have it is good. 

> 2. I have the search results display in a separate section to the post feed, I assume this still meets this criteria? "User can search the post content feed".

This still meets the criteria.

> 3. Should I try to break down some of my function further? Especially in reference to my [post_cards.mjs](https://github.com/Anclagen/SoSocial/blob/JS2/js/modules/render/post_cards.mjsl) where I have a very large function to build my posts and add all the relevant listeners and functionality, I have split some of the code for the forms, and replies into other files.

Yes, you should try abstracting some of this into new files. 

> 4. Error reporting with try catch blocks should I be placing these in the primary fetch functions or further down the chain where I use function to apply this to the page, or is it ok to nest multiple try catches by adding one at each stage?

Whenever you're dealing with the API will be a good place to use a `try catch` block. You can have them in your main API functions, but also in the individual functions if you're dealing with more API calls.

File: js/modules/render/loader.mjs (hunk 1 above)
Comment (Anclagen) [reply js/modules/render/loader.mjs]: Loader now says "Getting posts, please wait..."

**Pull request merged**